File: 프로그래머스/1/87389. 나머지가 1이 되는 수 찾기/나머지가 1이 되는 수 찾기.js
function solution(n) {
    for (let x = 1; x < n; x++) {
        if (n % x === 1) {
            return x;
        }
    }
    return -1; // 안전장치, 실제로는 여기까지 오지 않음
}

// 테스트
console.log(solution(10)); // 3
console.log(solution(12)); // 11

File: 프로그래머스/1/87389. 나머지가 1이 되는 수 찾기/나머지가 1이 되는 수 찾기.kt
class Solution {
    fun solution(n: Int): Int {
        for (x in 1 until n) {
            if (n % x == 1) {
                return x
            }
        }
        return -1 // 실제로는 여기까지 오지 않음
    }
}

fun main() {
    val sol = Solution()
    println(sol.solution(10)) // 3
    println(sol.solution(12)) // 11
}
